app/javascript/services/instrument.service.js

```diff
@@ -0,0 +1,32 @@
+const AudioContext = window.AudioContext || window.webkitAudioContext;
+
+export default class InstrumentService {
+    constructor(sample) {
+        this.sample = sample
+    }
+
+    play(rate) {
+        const context = new AudioContext();
+        const source = context.createBufferSource();
+        source.buffer = this.sample;
+        source.playbackRate.value = rate;
+        source.connect(context.destination);
+        source.start(0);
```

Comment:
An AudioContext has a `clock` 🕐  built in. When you instantiate a new AudioContext, the clock will start to tik.

When calling `source.start(0)`, it means "Start playing the sample as if it started when the clock was at 0.

If you say `source.start(10)`, it means "Start playing when the AudioContext clock will reach the time `10`".

In order to play instanteniously, we need to first ask the audio context: "What time is it?"

We can do that like this:

```suggestion
        const time = this.audioCtx.currentTime;
        source.start(time);
```

Then you can play `source.start(time)`. 

I just understood that while debugging my WaveTable engine :P

Reply:
Updated at [81eb0e6](https://github.com/malparty/nimbleat/pull/16/commits/81eb0e68581f202da3f65ca436380cc260e64416)